### PR TITLE
Bump kotest

### PR DESCRIPTION
**Pull request by aballano**

### CODE REVIEW

File: arrow-core-data/src/test/kotlin/arrow/core/AndThenTest.kt [hunk 1]
```diff
@@ -26,9 +26,9 @@ import arrow.typeclasses.EqK
 import arrow.typeclasses.EqK2
 import arrow.typeclasses.conest
 import arrow.typeclasses.counnest
-import io.kotlintest.properties.Gen
-import io.kotlintest.properties.forAll
-import io.kotlintest.shouldBe
+import io.kotest.properties.Gen
+import io.kotest.properties.forAll
```
Comment (rachelcarmena) [inline arrow-core-data/src/test/kotlin/arrow/core/AndThenTest.kt:30]: I think this package changes to `property`. For instance: https://github.com/kotest/kotest/blob/master/kotest-property/src/commonMain/kotlin/io/kotest/property/Gen.kt

File: arrow-core-data/build.gradle [hunk 2]
```diff
@@ -21,7 +21,7 @@ dependencies {
     compileOnly project(":arrow-meta")
     testCompileOnly project(":arrow-meta")
     testRuntime "org.junit.vintage:junit-vintage-engine:$JUNIT_VINTAGE_VERSION"
-    testCompile "io.kotlintest:kotlintest-runner-junit5:$KOTLIN_TEST_VERSION", excludeArrow
+    testCompile "io.kotest:kotest-runner-junit5-jvm:$KOTLIN_TEST_VERSION", excludeArrow
```
Comment (rachelcarmena) [inline arrow-core-data/build.gradle:24]: I think it's just necessary to remove `excludeArrow` here.

File: arrow-core-test/build.gradle [hunk 3]
```diff
@@ -16,6 +16,6 @@ dependencies {
     compile "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$KOTLIN_VERSION"
     compile "org.jetbrains.kotlinx:kotlinx-coroutines-core:$KOTLINX_COROUTINES_VERSION"
     testRuntime "org.junit.vintage:junit-vintage-engine:$JUNIT_VINTAGE_VERSION"
-    compile "io.kotlintest:kotlintest-runner-junit5:$KOTLIN_TEST_VERSION", excludeArrow
+    compile "io.kotest:kotest-runner-junit5-jvm:$KOTLIN_TEST_VERSION", excludeArrow
```
Comment (rachelcarmena) [inline arrow-core-test/build.gradle:19]: I'd just upgrade `arrow-core-test` which is using KotlinTest to **compile**. 

I think all the necessary dependencies should be:

```   
    compile "io.kotest:kotest-runner-junit5-jvm:$KOTEST_VERSION" 
    compile "io.kotest:kotest-assertions-core-jvm:$KOTEST_VERSION"
    compile "io.kotest:kotest-property-jvm:$KOTEST_VERSION"
    compile "io.kotest:kotest-runner-console-jvm:$KOTEST_VERSION"`
```

`excludeArrow` is not necessary because **Kotest** doesn't show Arrow as a transitive dependency.

File: arrow-core/build.gradle [hunk 4]
```diff
@@ -21,6 +21,6 @@ dependencies {
     compileOnly project(":arrow-meta")
     testCompileOnly project(":arrow-meta")
     testRuntime "org.junit.vintage:junit-vintage-engine:$JUNIT_VINTAGE_VERSION"
-    testCompile "io.kotlintest:kotlintest-runner-junit5:$KOTLIN_TEST_VERSION", excludeArrow
+    testCompile "io.kotest:kotest-runner-junit5-jvm:$KOTLIN_TEST_VERSION", excludeArrow
```
Comment (rachelcarmena) [inline arrow-core/build.gradle:24]: I think it's just necessary to remove `excludeArrow` here.

File: arrow-meta/build.gradle [hunk 5]
```diff
@@ -31,7 +31,7 @@ dependencies {
     testCompile project(":arrow-meta:arrow-meta-test-models")
     testCompile("junit:junit:$JUNIT_VERSION")
     testRuntime("org.junit.vintage:junit-vintage-engine:$JUNIT_VINTAGE_VERSION")
-    testCompile "io.kotlintest:kotlintest-runner-junit4:$KOTLIN_TEST_VERSION", excludeArrow
+    testCompile "io.kotest:kotest-runner-junit5-jvm:$KOTLIN_TEST_VERSION", excludeArrow
```
Comment (rachelcarmena) [inline arrow-meta/build.gradle:34]: I think it's just necessary to remove `excludeArrow` here because `testCompile` dependency.

File: arrow-syntax/build.gradle [hunk 6]
```diff
@@ -17,6 +17,6 @@ dependencies {
     compile project(":arrow-annotations")
     compile project(":arrow-core")
     testRuntime "org.junit.vintage:junit-vintage-engine:$JUNIT_VINTAGE_VERSION"
-    testCompile "io.kotlintest:kotlintest-runner-junit5:$KOTLIN_TEST_VERSION", excludeArrow
+    testCompile "io.kotest:kotest-runner-junit5-jvm:$KOTLIN_TEST_VERSION", excludeArrow
```
Comment (rachelcarmena) [inline arrow-syntax/build.gradle:20]: I think it's just necessary to remove `excludeArrow` here because `testCompile` dependency.

File: build.gradle [hunk 7]
```diff
@@ -29,4 +29,10 @@ configure(subprojects - project("arrow-meta")) {
     dependencies {
         signature 'net.sf.androidscents.signature:android-api-level-21:5.0.1_r2@signature'
     }
+
+    tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
+        kotlinOptions {
+            jvmTarget = "1.8"
+        }
+    }
```
Comment (rachelcarmena) [inline build.gradle:37]: Awesome job in this PR :clap: Please, could you locate these lines in `arrow` repository? `subprojects` section in `generic-conf.gradle`. `JAVA_VERSION` is available. Thanks!!

**Comment by rachelcarmena** [comment]
@aballano, the checks are going to find https://github.com/arrow-kt/arrow-fx/tree/ab/bump-kotest branch because https://github.com/arrow-kt/arrow-fx/pull/230 is open. Close that PR in order not to consider that branch during checks (I think it's not fixed yet).

File: arrow-core-data/src/main/kotlin/arrow/core/Nullable.kt [hunk 8]
```diff
@@ -1,9 +1,6 @@
 @file:Suppress("NAME_SHADOWING")
 package arrow.core
 
-inline fun <A, R> mapN(a: A?, fn: (A) -> R): R? =
```
Comment (nomisRev) [inline arrow-core-data/src/main/kotlin/arrow/core/Nullable.kt:4]: Why these changes unrelated to PR?

Comment (aballano) [reply arrow-core-data/src/main/kotlin/arrow/core/Nullable.kt:4]: https://github.com/arrow-kt/arrow-core/pull/189#pullrequestreview-452433217

**Comment by nomisRev** [comment]
@aballano Can we consider [this ticket](https://github.com/arrow-kt/arrow/issues/2138) in TODO with this PR?
